lupa/tests/test.py

```diff
@@ -2619,6 +2619,119 @@ def test_nil_debug_traceback(self):
+        super(TestOverflow, self).tearDown()
+
+    def test_no_overflow(self):
+        self.assertMathType(0, 'integer')
+        self.assertMathType(10, 'integer')
+        self.assertMathType(-10, 'integer')
+        self.assertMathType(self.maxinteger, 'integer')
+        self.assertMathType(-self.maxinteger, 'integer')
+        self.assertMathType(0.0, 'float')
+        self.assertMathType(-0.0, 'float')
+        self.assertMathType(10.0, 'float')
+        self.assertMathType(-10.0, 'float')
+        self.assertMathType(3.14, 'float')
+        self.assertMathType(-3.14, 'float')
+        self.assertMathType(self.maxfloat, 'float')
+        self.assertMathType(-self.maxfloat, 'float')
+
+    def assertMathType(self, number, math_type):
+        self.assertEqual(self.lua_type(number), 'number')
+        if self.lua_math_type is not None:
+            self.assertEqual(self.lua_math_type(number), math_type)
+
+class TestOverflowWithoutHandler(TestOverflow, unittest.TestCase):
+    lua_runtime_kwargs = dict(overflow_handler=None)
+
+    def test_overflow(self):
+        self.assertRaises(OverflowError, self.assertMathType, self.biginteger, 'integer')
+        self.assertRaises(OverflowError, self.assertMathType, int(self.maxfloat), 'integer')
+        self.assertRaises(OverflowError, self.assertMathType, self.bigfloat, 'integer')
+
+class TestOverflowWithFloatHandler(TestOverflow, unittest.TestCase):
+    lua_runtime_kwargs = dict(overflow_handler=float)
+
+    def test_overflow(self):
+        self.assertMathType(self.biginteger, 'float')
+        self.assertMathType(int(self.maxfloat), 'float')
+        self.assertRaises(OverflowError, self.assertMathType, self.bigfloat, 'float')
+
+class TestOverflowWithObjectHandler(TestOverflow, unittest.TestCase):
+
+    def test_overflow(self):
+        self.lua.execute('python.set_overflow_handler(function(o) return o end)')
+        self.assertEqual(self.lua.eval('type')(int(self.maxfloat)), 'userdata')
+
+class TestBadOverflowHandlerInPython(unittest.TestCase):
+    def test_error(self):
+        self.assertRaises(ValueError, lupa.LuaRuntime, overflow_handler=123)
+
+class TestBadOverflowHandlerInLua(SetupLuaRuntimeMixin, unittest.TestCase):
+    def test_number(self):
+        self._test_set_overflow_handler('123')
+
+    def test_table(self):
+        self._test_set_overflow_handler('{}')
+
+    def test_userdata(self):
+        self._test_set_overflow_handler('python.builtins.float')
+
+    def test_boolean(self):
+        self._test_set_overflow_handler('true')
+        self._test_set_overflow_handler('false')
+
+    def test_string(self):
+        self._test_set_overflow_handler('"abc"')
+
+    def test_thread(self):
+        self._test_set_overflow_handler('coroutine.create(function() end)')
+
+    def _test_set_overflow_handler(self, overflow_handler_code):
+        self.assertRaises(lupa.LuaError, self.lua.execute, 'python.set_overflow_handler(%s)' % overflow_handler_code)
```

Comment:
I would move these main test 'helper' functions up to the top. Classes are usually read from top to bottom, so this would help in understanding what is going on in this class.

Reply:
Resolved in https://github.com/scoder/lupa/pull/168/commits/1f36c0f05f803a3eee06b316c599d84c7625f372